### ISSUE TITLE
Can spark do URL rewriting / remapping to static resources? Or returning a static resource from get/post?

### ISSUE THREAD

**Issue by dessalines** [perwendel/spark#249]
For example, I'm using spark to serve static web pages as well, all of which are accessible at URLs like:
`http://my-ip:4567/web/html/home.html`
But I'd like to remap the following line to the one above
`http//my-ip:4567/home`

And of course, the other main one, is remapping parameter values to URLs:
`http://www.example.com/Blog/Posts.php?Year=2006&Month=12&Day=19`

could be mapped to:
`http://www.example.com/Blog/2006/12/19/`

Basically, how do I return a static resource with get or post?

**Comment by tchoulihan** [comment]
I figured out a workaround: simply read the static resource into a string, and serve it up. So something like:

```java
		get("/home", (req, res) -> {
			byte[] encoded = java.nio.file.Files.readAllBytes(Paths.get(path));
			s = new String(encoded, Charset.defaultCharset());
                        return s;
		});
```

**Comment by bahmanm** [comment]
@tchoulihan Why not simply let nginx serve the static resources?

**Comment by tchoulihan** [comment]
1) I don't use nginx, I prefer apache
2) I'd like to avoid them entirely anyway, because I'd like to keep the jar portable and have it do everything without having to rely on the webserver installed on the machine.

**Comment by jkeam** [comment]
nginx would let you easily cache those resources too.

**Comment by chillenious** [comment]
Whether or not nginx might be better, it's not what he's asking :-) I think
there's something to say for keeping everything in a single portable jar.

**Comment by jkeam** [comment]
haha yea, thats why I deleted my comment.  i wonder if there is a way to get this done out of the box without the work around.  i don't know off hand.

**Comment by amarseillan** [comment]
Define a route /home which forwards the request to the static file location? (Didn't try it, just wondering...)

**Comment by tchoulihan** [comment]
@amarseillan how would you do that?

**Comment by amarseillan** [comment]
sweared spark had a .forward method =P
tried to do it with the raw HttpServletRequest but it fails miserably

I don't think there's a way of doing this, will keep looking tho
